test/unit/plugin/v2/installer_test.rb

```diff
@@ -0,0 +1,414 @@
+require 'minitest/autorun'
+require 'minitest/test'
+# Other unit tests include the webmock framework, which is process-wide.
+# We need to disable it, or else mock many, many rubygems API calls.
+require 'webmock/minitest'
+
+require 'fileutils'
+require 'json'
+require_relative '../../../../lib/inspec/plugin/v2'
+require_relative '../../../../lib/inspec/plugin/v2/installer'
+
+require 'byebug'
+
+module InstallerTestHelpers
+  def reset_globals
+    ENV['HOME'] = @orig_home
+    ENV['INSPEC_CONFIG_DIR'] = nil
+    @installer.__reset
+  end
+
+  def copy_in_config_dir(fixture_name)
+    src = Dir.glob(File.join(@config_dir_path, fixture_name, '*'))
+    dest = File.join(@config_dir_path, 'empty')
+    src.each { |path| FileUtils.cp_r(path, dest) }
+  end
+
+  def setup
+    @orig_home = Dir.home
+
+    repo_path = File.expand_path(File.join( __FILE__, '..', '..', '..', '..', '..'))
+    mock_path = File.join(repo_path, 'test', 'unit', 'mock')
+    @config_dir_path = File.join(mock_path, 'config_dirs')
+    @plugin_fixture_src_path = File.join(mock_path, 'plugins', 'inspec-test-fixture')
+    @plugin_fixture_pkg_path = File.join(@plugin_fixture_src_path, 'pkg')
+
+    # This is unstable under CI; see https://github.com/inspec/inspec/issues/3355
+    # @ruby_abi_version = (RUBY_VERSION.split('.')[0,2] << '0').join('.')
+    @ruby_abi_version = (Gem.ruby_version.segments[0, 2] << 0).join('.')
+
+    @installer = Inspec::Plugin::V2::Installer.instance
+    reset_globals
+    WebMock.disable_net_connect!(allow: 'api.rubygems.org')
+  end
+
+  def teardown
+    reset_globals
+
+    # We use the 'empty' config dir for exercising a lot of installs.
+    # Purge it after every test.
+    unless ENV['INSPEC_TEST_PRESERVE_PLUGIN']
+      Dir.glob(File.join(@config_dir_path, 'empty', '*')).each do |path|
+        next if path.end_with? '.gitkeep'
+        FileUtils.rm_rf(path)
+      end
+    end
+
+    # TODO: may need to edit the $LOAD_PATH, if it turns out that we need to "deactivate" gems after installation
+  end
+end
+
+#-----------------------------------------------------------------------#
+# basics
+#-----------------------------------------------------------------------#
+class PluginInstallerBasicTests < MiniTest::Test
+  include InstallerTestHelpers
+
+  # it's a singleton
+  def test_it_should_be_a_singleton
+    klass = Inspec::Plugin::V2::Installer
+    assert_equal klass.instance, klass.instance, "Calling instance on the Installer should always return the same object"
+    assert_raises(NoMethodError, 'Installer should have a private constructor') { klass.new }
+  end
+
+  # it should know its gem path
+  def test_it_should_know_its_gem_path_with_a_default_location
+    ENV['HOME'] = File.join(@config_dir_path, 'fakehome')
+    expected = File.join(ENV['HOME'], '.inspec', 'gems', @ruby_abi_version)
+    assert_equal expected, @installer.gem_path
```

Comment:
Same regarding arguments in `()`...though this is not linted nor in our style guide yet...I will ignore future "violations".

Reply:
I think we might disagree on this. As a team we decided to highlight the style issues in the tests here:

https://github.com/inspec/inspec/pull/3381

No reason we shouldn't do it internally as well. I'll bring it up internally.

Reply:
I brought it up (which you know because you were there...but posterity and what not)

TL:DR: Despite #3381, we should not limit contributions in such a fashion. I (we) will avoid such feedback in the future.